tests/unit/cli/cli-test.js

```diff
@@ -228,43 +254,78 @@ describe('Unit: CLI', function() {
     });
   });
 
-  describe('command interruption handler', function() {
-    let onCommandInterrupt;
+  describe('command interruption', function() {
+    let interruptionHandeled;
     beforeEach(function() {
-      onCommandInterrupt = td.matchers.isA(Function);
+      interruptionHandeled = false;
     });
 
-    it('sets up handler before command run', function() {
-      const CustomCommand = Command.extend({
-        name: 'custom',
+    const FakeCommand = Command.extend({
+      name: 'fake',
 
-        beforeRun() {
-          td.verify(willInterruptProcess.addHandler(onCommandInterrupt));
+      beforeRun() {
+        return Promise.resolve();
+      },
 
-          return Promise.resolve();
-        },
+      run() {
+        return new Promise(resolve => {
+          setTimeout(() => resolve(), 50);
+        });
+      },
 
-        run() {
-          return Promise.resolve();
+      onInterrupt() {
+        return new Promise(resolve => {
+          setTimeout(() => {
+            interruptionHandeled = true;
+            resolve();
+          }, 100); // ensure we cleanup longer than command run executed
+        });
+      },
+    });
+
+    it('sets up handler before command run', function() {
+      registerCommand(FakeCommand.extend({
+        beforeRun() {
+          _process.emit('SIGINT');
         },
+      }));
+
+      return ember(['fake']).finally(function() {
+        expect(interruptionHandeled).to.equal(true);
       });
+    });
 
-      project.eachAddonCommand = function(callback) {
-        callback('custom-addon', {
-          CustomCommand,
-        });
-      };
+    it('cleans up handler right after command is finished', function() {
+      registerCommand(FakeCommand.extend({
+        onInterrupt() {
+          interruptionHandeled = true;
+        },
+      }));
 
-      return ember(['custom']);
+      return ember(['fake']).finally(function() {
+        _process.emit('SIGINT');
+
+        // ensure interruption handler has enough time to be triggered
+        return new Promise(resolve => setTimeout(resolve, 50));
+      }).then(() => {
+        expect(interruptionHandeled).to.equal(false);
+      });
     });
 
-    it('cleans up handler after command finished', function() {
-      stubValidateAndRun('serve');
+    it(`rejected with a proper error on crash`, function() {
+      const error = new SilentError('OMG');
 
-      return ember(['serve']).finally(function() {
-        td.verify(willInterruptProcess.removeHandler(onCommandInterrupt));
+      registerCommand(FakeCommand.extend({
+        run() {
+          throw error;
+        },
+      }));
+
+      return expect(ember(['fake'])).to.be.rejectedWith(error).then(() => {
+        expect(interruptionHandeled).to.equal(false);
```

Comment:
FYI the [original](https://github.com/ember-cli/ember-cli/pull/7019/files#diff-aa94f362c4d6ae97d4d491b3a8cc4887R256) test tried to check that cleanup happens on error.  Anyway there is no such logic in cli.
Do we really want to cleanup if every time command throws an exception?